assets/js/googlesitekit/modules/datastore/sharing-settings.test.js

```diff
@@ -0,0 +1,367 @@
+/**
+ * `googlesitekit/modules` datastore: sharing settings tests.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { CORE_MODULES } from './constants';
+import {
+	createTestRegistry,
+	unsubscribeFromAll,
+	untilResolved,
+} from '../../../../../tests/js/utils';
+import FIXTURES from './__fixtures__';
+import { MODULES_SEARCH_CONSOLE } from '../../../modules/search-console/datastore/constants';
+import { MODULES_PAGESPEED_INSIGHTS } from '../../../modules/pagespeed-insights/datastore/constants';
+
+describe( 'core/modules sharing-settings', () => {
+	const sharingSettings = {
+		settings: {
+			'search-console': {
+				sharedRoles: [ 'editor', 'subscriber' ],
+				management: 'all_admins',
+			},
+			analytics: {
+				sharedRoles: [ 'editor' ],
+				management: 'owner',
+			},
+			'pagespeed-insights': {
+				sharedRoles: [ 'editor' ],
+				management: 'all_admins',
+			},
+		},
+	};
+
+	let registry;
+	let store;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		store = registry.stores[ CORE_MODULES ].store;
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'setSharingManagement', () => {
+			const settingsWithoutManagement = {
+				'search-console': {
+					sharedRoles: [ 'editor', 'subscriber' ],
+				},
+				analytics: {
+					sharedRoles: [ 'editor' ],
+				},
+				'pagespeed-insights': {
+					sharedRoles: [ 'editor' ],
+				},
+			};
+
+			it( 'requires the moduleSlug param', () => {
+				expect( () => {
+					registry.dispatch( CORE_MODULES ).setSharingManagement();
+				} ).toThrow( 'moduleSlug is required' );
+			} );
+
+			it( 'requires the management param', () => {
+				expect( () => {
+					registry
+						.dispatch( CORE_MODULES )
+						.setSharingManagement( 'analytics' );
+				} ).toThrow( 'management must be one of: all_admins, owner.' );
+			} );
+
+			it( 'receives management and sets it to the sharing settings modules', () => {
```

Comment:
This test seems to be overly complicated. All what we need to do here are these 3 steps:

1. Set initial sharing settings using the `receiveGetSharingSettings` selector.
2. Dispatch the `setSharingManagement` action for the `analytics` module.
3. Check that `store.getState().sharingSettings.analytics.management` equals to passed value.